### PR TITLE
Fixed need follow up tagged applications to be automatically reminded

### PR DESCRIPTION
**Pull request by Bahugunajii**
### Target
#1592 

### Description
- Created a mail functionality that will be sent to the candidates if an application has a `Need follow up` tag associated with it 
- As the mail is sent, an entry of the applicant will be made  automatically in the hr_follow_ups table 

### Checklist:
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php [hunk 1]
```diff
@@ -59,6 +59,17 @@ public function run()
             'setting_key' => 'applicant_verification_subject',
             'setting_value' => 'Confirm your email address',
         ]);
+        Setting::updateOrCreate([
+            'module' => 'hr',
+            'setting_key' => 'Follow_up_email_for_scheduling_interview_subject',
```
Comment (GauravGusain98) [inline Modules/HR/Database/Seeders/SettingsTableSeeder.php:64]: key should be in lowercase
```suggestion
            'setting_key' => 'follow_up_email_for_scheduling_interview_subject',
```

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php [hunk 2]
```diff
@@ -59,6 +59,17 @@ public function run()
...
+            'setting_value' => 'Follow up email for scheduling interview ',
+        ]);
+        Setting::updateOrCreate([
+            'module' => 'hr',
+            'setting_key' => 'Follow_up_email_for_scheduling_interview_body',
```
Comment (GauravGusain98) [inline Modules/HR/Database/Seeders/SettingsTableSeeder.php:69]:
```suggestion
            'setting_key' => 'follow_up_email_for_scheduling_interview_body',
```

Comment (GauravGusain98) [reply Modules/HR/Database/Seeders/SettingsTableSeeder.php:69]: Also, we should not use hardcoded values. As I can see this key being used at many place so can we take the value of this key from the config file inside the HR Module. 

For reference you can see the invoice templates config which is stored in config file of Invoice module.

File: Modules/HR/Listeners/FollowUpListener.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @param  object  $event
+     * @return void
+     */
+    public function handle($event)
+    {
+        $application = $event->application;
+        $applicant = $application->applicant;
+
+        $subject = Setting::where('module', 'hr')->where('setting_key', 'Follow_up_email_for_scheduling_interview_subject')->first();
+        $body = Setting::where('module', 'hr')->where('setting_key', 'Follow_up_email_for_scheduling_interview_body')->first();
```
Comment (GauravGusain98) [inline Modules/HR/Listeners/FollowUpListener.php:35]: same here

File: Modules/HR/Listeners/FollowUpListener.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        $application->update([
+            'applicant_verification_subject' => $subject->setting_value,
+            'applicant_verification_body' => $body->setting_value,
+        ]);
+
+        FollowUp::create([
+            'hr_application_round_id' => $application->latestApplicationRound->id,
+            'comments' => 'sent follow up email',
+            'conducted_by' => User::select('id')->first()->id
+                ]);
```
Comment (GauravGusain98) [inline Modules/HR/Listeners/FollowUpListener.php:52]:
```suggestion
        FollowUp::create([
            'hr_application_round_id' => $application->latestApplicationRound->id,
            'comments' => 'sent follow up email',
            'conducted_by' => User::select('id')->first()->id
        ]);
```

File: resources/views/settings/hr/follow-up.blade.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+						<div class="form-group">
+							<label for="setting_key[Follow_up_email_for_scheduling_interview_subject]">Subject</label>
+							<input type="text" name="setting_key[Follow_up_email_for_scheduling_interview_subject]" class="form-control" value="{{ isset($settings['Follow_up_email_for_scheduling_interview_subject']->setting_value) ? $settings['Follow_up_email_for_scheduling_interview_subject']->setting_value : '' }}">
+						</div>
+					</div>
+				</div>
+				<div class="form-row">
+					<div class="col-md-12">
+						<div class="form-group">
+							<label for="setting_key[Follow_up_email_for_scheduling_interview_body]">Mail body:</label>
```
Comment (GauravGusain98) [inline resources/views/settings/hr/follow-up.blade.php:19]: Take value from config

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Minor suggestion. Please check

File: resources/views/settings/hr/follow-up.blade.php (hunk 5 above)
Comment (Bahugunajii) [reply resources/views/settings/hr/follow-up.blade.php]: Done!

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php (hunk 2 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Seeders/SettingsTableSeeder.php]: Done!

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php (hunk 1 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Seeders/SettingsTableSeeder.php]: Done!

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php [hunk 6]
```diff
@@ -59,6 +59,17 @@ public function run()
             'setting_key' => 'applicant_verification_subject',
             'setting_value' => 'Confirm your email address',
         ]);
+        Setting::updateOrCreate([
+            'module' => 'hr',
+            'setting_key' => 'follow_up_email_for_scheduling_interview_subject',
```
Comment (GauravGusain98) [inline Modules/HR/Database/Seeders/SettingsTableSeeder.php:64]: We can also take it from config

File: Modules/HR/Database/Seeders/SettingsTableSeeder.php [hunk 7]
```diff
@@ -59,6 +59,17 @@ public function run()
...
+            'setting_value' => 'Follow up email for scheduling interview ',
+        ]);
+        Setting::updateOrCreate([
+            'module' => 'hr',
+            'setting_key' => 'follow_up_email_for_scheduling_interview_body',
```
Comment (GauravGusain98) [inline Modules/HR/Database/Seeders/SettingsTableSeeder.php:69]: same here

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Minor changes requested. Please check.

**Review by GauravGusain98 — APPROVED** [review]
Looks good

**Pull request merged**